### PR TITLE
SN-15 Create Conversation Model & model users_conversation

### PR DESCRIPTION
**Pull request by Slavahaim**
1. Create Conversation Model 
2. Create model users_conversation
3. added factories bot
4. added spec

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ AllCops:
     - 'Gemfile.lock'
     - 'config.ru'
     - 'db/migrate/20220120101655_create_active_storage_tables.active_storage.rb'
```
Comment (PFDeve) [inline .rubocop.yml]: I would remove this line if you added `'db/**/*'`

Comment (ramantechart) [reply .rubocop.yml]: Such excludes should be done in .rubocop-todo.yml files

File: app/models/user.rb [hunk 2]
```diff
@@ -20,5 +20,8 @@ class User < ApplicationRecord
   validates :net_state, presence: true
   validates :profile_id, presence: true, uniqueness: true
 
+  has_many :users_conversations
+  has_many :conversations, through: :users_conversations
```
Comment (skripchenkoVeta) [inline app/models/user.rb]: Please use correct order

File: spec/factories/conversations.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_conversations_on_name  (name) UNIQUE
+#
+FactoryBot.define do
+  factory :conversation do
+    name { Faker::Lorem.word }
+  end
```
Comment (skripchenkoVeta) [inline spec/factories/conversations.rb]: Please add more traits :)

File: spec/models/conversation_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# Indexes
+#
+#  index_conversations_on_name  (name) UNIQUE
+#
+require 'rails_helper'
+
+RSpec.describe Conversation, type: :model do
+  subject(:conversation) { FactoryBot.build(:conversation) }
+
+  describe 'validation test' do
```
Comment (skripchenkoVeta) [inline spec/models/conversation_spec.rb:21]: I think it's not quite right
Usually describe is used for entities (who? what?)
For example: describe _'#user'_ , describe _'associations'_
You can read about it here: https://www.betterspecs.org/#describe

File: spec/models/users_conversation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#  fk_rails_...  (user_id => users.id)
+#
+require 'rails_helper'
+
+RSpec.describe UsersConversation, type: :model do
+  let(:conversation) { FactoryBot.build(:conversation) }
+  let(:user) { FactoryBot.build(:user) }
+  let(:users_conversation) { FactoryBot.create(:users_conversation, conversation: conversation, user: user) }
+
+  describe 'validation test' do
```
Comment (skripchenkoVeta) [inline spec/models/users_conversation_spec.rb:30]: same

File: spec/models/conversation_spec.rb (hunk 4 above)
Comment (Slavahaim) [reply spec/models/conversation_spec.rb:21]: conversation  is the essence. 
https://www.betterspecs.org/#describe -- I read, in general, according to the recommendations. But I'll take care of it )

File: spec/models/users_conversation_spec.rb (hunk 5 above)
Comment (Slavahaim) [reply spec/models/users_conversation_spec.rb:30]: same

File: app/models/user.rb (hunk 2 above)
Comment (Slavahaim) [reply app/models/user.rb]: OK!

File: spec/factories/conversations.rb (hunk 3 above)
Comment (Slavahaim) [reply spec/factories/conversations.rb]: What traits do you mean?

Comment (skripchenkoVeta) [reply spec/factories/conversations.rb]: For example:
In your spec/models/conversation_spec.rb file
On line 24 there is a line
conversation.name = nil
take it out to trait and then use

File: .rubocop.yml (hunk 1 above)
Comment (Slavahaim) [reply .rubocop.yml]: I changed

File: spec/models/conversation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    context 'when we have required fields to fill in' do
+      it 'ensures name presecence' do
+        conversation.name = nil
+        expect(conversation.save).to eq(false)
```
Comment (ramantechart) [inline spec/models/conversation_spec.rb:25]: It doesn't check AR error, but it should.

File: spec/models/conversation_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+#
+#  index_conversations_on_name  (name) UNIQUE
+#
+require 'rails_helper'
+
+RSpec.describe Conversation, type: :model do
+  subject(:conversation) { FactoryBot.build(:conversation) }
+
+  describe 'validation test' do
+    context 'when we have required fields to fill in' do
```
Comment (ramantechart) [inline spec/models/conversation_spec.rb:22]: this title is wrong.
It says we have fields, but later name is nil.
It's confusing

Comment (ramantechart) [reply spec/models/conversation_spec.rb:22]:
```
describe 'invalid conversation' do
  context 'when missing name attribute' do
    let(:nameless_conversation) { build(:conversation, :nameless) }
    it 'rises validation error' do
      expect(conversation.save!).to raise(ARError, "errror message")
    end
  end
end
```

**Pull request merged**